addons/hr_payroll/models/hr_benefit.py

```diff
@@ -133,34 +122,53 @@ def compute_conflicts_leaves_to_approve(self):
             })
         return bool(conflicts)
 
-    def safe_duplicate_create(self, vals):
+    def _safe_duplicate_create(self, vals_list):
         """
-        Create benefit but silently abord if it already exists in the database (according to the _unique constraints).
-        @return: new record if it didn't exist, empty recordset otherwise.
+        Create benefit but silently abord if it already exists in the database.
+        @return: new record id if it didn't exist.
         """
-        try:
-            with mute_logger('odoo.sql_db'), self.env.cr.savepoint():
-                # Any database call will be run inside a postgresql savepoint which is
-                # rolledback if an exception occurs allowing to make subsequent calls in the transaction.
-                # (Otherwise InternalError is raised by any subsequent database operations)
-                return self.create(vals)
-        except IntegrityError as err:
-            if 'hr_benefit__unique' not in err.pgerror:
-                raise err
-            return self.env['hr.benefit']
+        query = """
+                INSERT INTO hr_benefit (
+                    name,
+                    date_start,
+                    date_stop,
+                    duration,
+                    benefit_type_id,
+                    display_warning,
+                    employee_id,
+                    state,
+                    leave_id,
+                    active)
+                VALUES {}
+                ON CONFLICT DO NOTHING
+                RETURNING id""".format(','.join(['%s'] * len(vals_list)))
+        params = [(
+            vals['name'],
+            vals['date_start'],
+            vals['date_stop'],
+            self._get_duration(vals['date_start'], vals['date_stop']),
+            vals.get('benefit_type_id', False) or None,
+            not bool(vals['benefit_type_id']),
+            vals['employee_id'],
+            'confirmed',
+            vals.get('leave_id', False) or None,
+            True) for vals in vals_list]
+        self.env.cr.execute(query, params)
+        result = self.env.cr.fetchall()
+        return [res_id[0] for res_id in result] if result else False
```

Comment:
Could we try something like this, to keep the modularity:

    def _safe_duplicate_create(self, vals_list, start_date, stop_date):
        month_recs = self.search_read([('start_date', '>=', start_date), ('stop_date', '<=', stop_date)],
                                      ['employee_id', 'start_date', 'stop_date', 'benefit_type_id'])
        existing_entries = [(a,b,c,d) for r in month_recs]
        unique_vals = [v for v in vals_list if (v[''], v[''], v[''], v['']) not in existing_entries]
        return self.create(vals_list)